api/src/methods/forkchoice_updated.rs

```diff
@@ -72,14 +81,17 @@ async fn inner_execute_v3(
 
     // If `payload_attributes` are present then tell state to start producing a new block
     let payload_id = if let Some(attrs) = payload_attributes {
-        let (tx, rx) = oneshot::channel();
+        let payload_attributes = Payload::from(attrs);
+        let payload_id = payload_id.new_payload_id(
```

Comment:
Now that I think about it, it's a bit confusing to call this struct `payload_id`, when what it does is actually generating ids. Considering it's a trait, one obvious thing would be to call that trait itself `PayloadIder`, but it's ugly. Whereas `PayloadIdGenerator` is a bit of a mouthful. But I believe some more precise naming would be beneficial.

Reply:
I agree it should be clear that we passing in something with the capability to create payload ids as opposed to something that is itself already a payload_id.